src/components/Staking/StakeManagement/Unbonding/index.js

```diff
@@ -1,43 +1,91 @@
-import React from 'react';
+import React, { useEffect } from 'react';
 import PropTypes from 'prop-types';
-import { useSelector } from 'react-redux';
+import { useSelector, useDispatch } from 'react-redux';
 import { formatDollars } from '../../../../utils/walletHelpers';
 import { blockchainSelectors, validatorSelectors } from '../../../../redux/selectors';
-import { eraToDays } from '../../../../utils/staking';
+import { blockTimeFormatted } from '../../../../utils/staking';
+import { validatorActions } from '../../../../redux/actions';
 
-function UnbondingRow({ value, era }) {
-  const activeEra = useSelector(blockchainSelectors.activeEra);
-  const ready = activeEra.index.gte(era.unwrap());
+function UnbondingRow({ unlock, blocks }) {
+  return (
+    <li>
+      {formatDollars(unlock.value)}
+      {' '}
+      LLD
+      {' '}
+      {
+       `will unlock in ${blockTimeFormatted(blocks)}`
+      }
+    </li>
+  );
+}
 
+UnbondingRow.propTypes = {
+  // eslint-disable-next-line react/forbid-prop-types
+  unlock: PropTypes.object.isRequired,
+  // eslint-disable-next-line react/forbid-prop-types
+  blocks: PropTypes.object.isRequired,
+};
+
+function UnbondingRowReady({ value }) {
   return (
     <li>
       {formatDollars(value)}
       {' '}
       LLD
       {' '}
-      {ready ? 'ready to withdraw' : `will unlock on ${eraToDays(era)} Days`}
+      ready to withdraw
     </li>
   );
 }
 
-UnbondingRow.propTypes = {
+UnbondingRowReady.propTypes = {
   // eslint-disable-next-line react/forbid-prop-types
   value: PropTypes.object.isRequired,
-  // eslint-disable-next-line react/forbid-prop-types
-  era: PropTypes.object.isRequired,
 };
 
-export default function Unbonding() {
-  const info = useSelector(validatorSelectors.info);
+export default function Unbonding({ info }) {
+  const dispatch = useDispatch();
+  const stakingData = useSelector(validatorSelectors.stakingData);
+  const blockNumber = useSelector(blockchainSelectors.blockNumber);
+  const activeEra = useSelector(blockchainSelectors.activeEra);
+
+  useEffect(() => {
+    if (info.unlocking.length === 0) return;
+    dispatch(validatorActions.getStakingData.call());
+  }, [dispatch, blockNumber, info]);
 
-  if (!info || info.unlocking.length <= 0) return null;
+  if (
+    !info?.unlocking
+    || info?.unlocking.length === 0
+  ) return null;
 
   return (
     <>
       Currently unstaking:
       <ul>
-        {info.unlocking.map(({ value, era }) => <UnbondingRow key={era.toString()} {...{ value, era }} />)}
+        {
+        info.unlocking.map(({ era, value }) => (
+          activeEra.index.gte(era)
```

Comment:
```suggestion
          activeEra.index.gte(era.unwrap())
```

`activeEra.index` is just a number (wrapped in BN), but `era` is `Compact<u32>`. Era needs to be unwrapped, otherwise this always returns true (and we get duplicated rows).